staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -190,20 +219,39 @@ func ExtractList(obj runtime.Object) ([]runtime.Object, error) {
 			default:
 				list[i] = nil
 			}
-		case runtime.Object:
-			list[i] = item
+		case raw.Type() == objectType || raw.Type().Implements(objectType):
+			list[i] = raw.Interface().(runtime.Object)
 		default:
-			var found bool
-			if list[i], found = raw.Addr().Interface().(runtime.Object); !found {
+			if !alloc {
```

Comment:
same comment, move the `alloc` handling code to its own `case` before the `default:` case to leave the existing code as-is

Reply:
resolved,  PTAL~